### PR TITLE
add expat 2.2.9

### PR DESCRIPTION
**Pull request by ericLemanissier**
Specify library name and version:  **expat/2.2.9**

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SSE4** [comment]
for the record - the library is now named `libexpatd.lib` for some reason (it was just `libexpat.lib`)

**Comment by SSE4** [comment]
seems to be changed in: https://github.com/libexpat/libexpat/commit/1544741d0365e5a97680a94f4d97e41b38ccb6c6

**Comment by ericLemanissier** [comment]
this commit is also present on 2.2.8, and yet https://github.com/conan-io/conan-center-index/pull/310 passed ?

**Comment by uilianries** [comment]
> /usr/bin/ld: cannot find -llibexpat

It seems like the exported name is wrong, it should be `expat` only

File: recipes/expat/all/conanfile.py [hunk 1]
```diff
@@ -52,12 +51,14 @@ def package(self):
...
+        tools.rmdir(os.path.join(self.package_folder, "share"))
 
     def package_info(self):
         self.cpp_info.name = "EXPAT"
         if self.settings.os == "Windows" and self.settings.build_type == "Debug":
-            self.cpp_info.libs = ["expatd"]
+            self.cpp_info.libs = ["libexpatd"]
         else:
-            self.cpp_info.libs = ["expat"]
+            self.cpp_info.libs = ["libexpat"]
```
Comment (uilianries) [inline recipes/expat/all/conanfile.py]:
```suggestion
            self.cpp_info.libs = ["expat"]
```

https://github.com/conan-io/conan-center-index/pull/311#issuecomment-553620956

File: recipes/expat/all/conanfile.py [hunk 2]
```diff
@@ -52,11 +51,15 @@ def package(self):
...
         tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, "share"))
 
     def package_info(self):
         self.cpp_info.name = "EXPAT"
-        if self.settings.os == "Windows" and self.settings.build_type == "Debug":
-            self.cpp_info.libs = ["expatd"]
+        if self.settings.os == "Windows":
+            prefix = "lib" if self.version >= "2.2.9" else ""
```
Comment (SSE4) [inline recipes/expat/all/conanfile.py]: I hope this compares version as version, not as a string. @uilianries please confirm

Comment (ericLemanissier) [reply recipes/expat/all/conanfile.py]: that's how I interpreted https://github.com/conan-io/conan/blob/develop/conans/client/tools/version.py#L48

Comment (danimtb) [reply recipes/expat/all/conanfile.py]: not sure if that `self.version` is a Version class

**Comment by ericLemanissier** [comment]
I just rebased the changes on top of master, because it conflicted with 2.2.8 now that https://github.com/conan-io/conan-center-index/pull/310 is merged

File: recipes/expat/all/conanfile.py (hunk 2 above)
Comment (uilianries) [reply recipes/expat/all/conanfile.py]: IMO I prefer splitting or using `tools.collect_libs`. Generic recipes should not contain conditionals for specific versions.

**Pull request merged**